model/distro/distro_test.go

```diff
@@ -638,3 +638,36 @@ func TestGetAuthorizedKeysFile(t *testing.T) {
 		assert.Equal(t, expected, d.GetAuthorizedKeysFile())
 	})
 }
+
+func TestGetDistrosForImage(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	testConfig := testutil.TestConfig()
+	assert := assert.New(t)
+	session, _, err := db.GetGlobalSessionFactory().GetSession()
+	assert.NoError(err)
+	require.NotNil(t, session)
+	defer session.Close()
+	require.NoError(t, session.DB(testConfig.Database.DB).DropDatabase())
+
+	numCorrectDistros := 3
+	imageID := "distro"
+	otherImageID := "not_distro"
+	for i := 0; i < numCorrectDistros; i++ {
+		d := &Distro{
+			Id:      fmt.Sprintf("distro_%d", i),
+			ImageID: imageID,
+		}
+		assert.Nil(d.Insert(ctx))
+	}
+	d := &Distro{
+		Id:      fmt.Sprintf("distro_%d", rand.Int()),
+		ImageID: otherImageID,
+	}
+	assert.Nil(d.Insert(ctx))
```

Comment:
I'd prefer to write these all out explicitly, since it's easier to read:

```
	d1 := &Distro{
		Id:      "distro-1",
		ImageID: imageID,
	}
	assert.Nil(t, d1.Insert(ctx))

	d2 := &Distro{
		Id:      "distro-2",
		ImageID: imageID,
	}
	assert.Nil(t, d2.Insert(ctx))

        // etc....
```